hkmc2/shared/src/test/mlscript/ctx/ClassCtxParams.mls

```diff
@@ -1,40 +1,76 @@
 :js
-:todo
 
 
-class Foo()(using Int)
-//│ ╔══[ERROR] Multiple parameter lists are not supported for this definition.
-//│ ║  l.5: 	class Foo()(using Int)
-//│ ╙──     	      ^^^^^^^^^^^^^^^^
+using Int = 42
+
 
 class Foo(using val n: Int)
-//│ ╔══[ERROR] Illegal definition in term position.
-//│ ║  l.10: 	class Foo(using val n: Int)
-//│ ╙──      	                    ^^^^^^
-//│ ═══[ERROR] Expected a type symbol, got <error>
 
+:expect Foo(42)
+Foo
+//│ = Foo(42)
 
-fun summon[A](using a: A) = a
+:expect 42
+Foo.n
+//│ = 42
 
-class Foo(using Int) with
-  fun foo() = summon[Int]
 
+class Foo(using n: Int) with
+  fun foo = n
 
-fun summon(using a: Int) = a
+:expect Foo()
+Foo
+//│ = Foo()
 
-class Foo(using Int) with
-  fun foo() = summon
+:expect 42
+Foo.foo
+//│ = 42
 
-Foo()
-//│ = Foo()
 
+class Bar(using val s: Str)
+
+:e
+Bar
+//│ ╔══[ERROR] Cannot query instance of type Str for call: 
+//│ ║  l.33: 	Bar
+//│ ║        	^^^
+//│ ╟── Required by contextual parameter declaration: 
+//│ ║  l.30: 	class Bar(using val s: Str)
+//│ ║        	                    ^^^^^^
+//│ ╙── Missing instance: Expected: Str; Available: (Int)
+//│ = Bar(undefined)
+
+class Bar(using s: Str)
 
-fun foo(using Int) = 1
+:e
+Bar
+//│ ╔══[ERROR] Cannot query instance of type Str for call: 
+//│ ║  l.46: 	Bar
+//│ ║        	^^^
+//│ ╟── Required by contextual parameter declaration: 
+//│ ║  l.43: 	class Bar(using s: Str)
+//│ ║        	                ^^^^^^
+//│ ╙── Missing instance: Expected: Str; Available: (Int)
+//│ = Bar()
 
-using Int = 1
 
-:re
-foo()
-//│ ═══[RUNTIME ERROR] TypeError: tmp is not a function
 
+fun summon(using i: Int) = i
```

Comment:
This reminds me that in Scala, there is no special term for summon, it's a normal definition. Could we make `use` not a keyword and simply define it in Predef?

Reply:
Yes we have one https://github.com/hkust-taco/mlscript/issues/272